pkg/accounting/pricer.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package accounting
+
+import (
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type Pricer interface {
+	// PeerPrice is the price the peer charges for a given chunk hash
+	PeerPrice(peer, chunk swarm.Address) uint64
+	// Price is the price we charge for a given chunk hash
+	Price(chunk swarm.Address) uint64
+}
```

Comment:
I was thinking of `Pricer` more as a general source of price information for other components. Really the only thing retrieval protocol needs is the `PeerPrice` function which can be used to query the price for a chunk for a peer (including the node itself, the fact that this might be calculated by some strategy instead of being stored is not something the protocol needs to be aware of). The `Price` function only exists as a helper since the component using the `Pricer` might not know our own overlay address.

Anyway not much thought has been put into this yet. I doubt we can really finalize the interface until we have figured out the kind of pricing strategies we want.

Reply:
> I doubt we can really finalize the interface until we have figured out the kind of pricing strategies we want.

Agree